jina/docker/hubapi.py

```diff
@@ -1,112 +1,104 @@
-
 import json
-import requests
-from typing import Dict
+from typing import Dict, Sequence, Any, Optional
+from urllib.error import HTTPError
+from urllib.parse import urlencode
+from urllib.request import Request, urlopen
+
 from pkg_resources import resource_stream
 
 from .helper import credentials_file
 from ..helper import yaml, colored
+from ..logging.profile import TimeContext
 
 
-def _list(logger, name: str = None, kind: str = None, type_: str = None, keywords: tuple = ('numeric')):
-    """ Hub API Invocation to run `hub list` """
+def _list(logger, image_name: str = None, image_kind: str = None,
+          image_type: str = None, image_keywords: Sequence = ()) -> Optional[Dict[str, Any]]:
+    """ Use Hub api to get the list of filtered images
+
+    :param logger: logger to use
+    :param image_name:
+    :param image_kind:
+    :param image_type:
+    :param image_keywords:
+    :return: a dict of manifest specifications, each coresponds to a hub image
+    """
     # TODO: Shouldn't pass a default argument for keywords. Need to handle after lambda function gets fixed
     with resource_stream('jina', '/'.join(('resources', 'hubapi.yml'))) as fp:
         hubapi_yml = yaml.load(fp)
-    
-    hubapi_url = hubapi_yml['hubapi']['url']
-    hubapi_list = hubapi_yml['hubapi']['list']
-    params = {}
-    if name:
-        params['name'] = name
-    if kind:
-        params['kind'] = kind
-    if type_:
-        params['type'] = type_
-    if keywords:
-        # The way lambda function handles params, we need to pass them comma separated rather than in an iterable 
-        params['keywords'] = ','.join(keywords) if len(keywords) > 1 else keywords
+        hubapi_url = hubapi_yml['hubapi']['url'] + hubapi_yml['hubapi']['list']
+
+    params = {
+        'name': image_name,
+        'kind': image_kind,
+        'type': image_type,
+        'keywords': ','.join(image_keywords) if image_keywords else None
```

Comment:
`,'` should be prepended only if more than one keyword is specified. if it's just one keyword then its a single element list, no `,` needed

Reply:
default behavior wont hurt, cleaner code

```
Python 3.7.5 (default, Jan 19 2020, 12:44:36)
[Clang 11.0.0 (clang-1100.0.33.17)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> ','.join([])
''
```

Reply:
background: keywords specified in cli with `nargs` will always result a list